app/routes/events/components/RegistrationMeta.js

```diff
@@ -21,44 +26,104 @@ type Props = {
   hasSimpleWaitingList: boolean,
   useConsent: boolean,
   hasEnded: boolean,
+  photoConsents?: Array<PhotoConsent>,
+  eventSemester: EventSemester,
 };
 
 const ConsentStatus = ({
   useConsent,
-  photoConsent,
+  LEGACY_photoConsent,
   hasEnded,
+  photoConsents,
+  eventSemester,
 }: {
   useConsent: boolean,
-  photoConsent: EventRegistrationPhotoConsent,
+  LEGACY_photoConsent: LEGACY_EventRegistrationPhotoConsent,
   hasEnded: boolean,
+  photoConsents?: Array<PhotoConsent>,
+  eventSemester: EventSemester,
 }) => {
   if (!useConsent) return null;
-  switch (photoConsent) {
-    case 'PHOTO_NOT_CONSENT':
-      return (
-        <div>
-          <i className="fa fa-check-circle" /> Du har valgt <i>NEI</i> på
-          samtykke om bilder
-        </div>
+
+  const consentInfo = (className, consentDescription) => (
+    <div>
+      <i className={className} />
+      {' ' + consentDescription}
+    </div>
+  );
```

Comment:
Move this out of the outer function as well (and make it a component pls)

Reply:
✅